src/main/java/io/jenkins/plugins/credentials/secretsmanager/AwsCredentials.java

```diff
@@ -0,0 +1,173 @@
+package io.jenkins.plugins.credentials.secretsmanager;
+
+import com.cloudbees.jenkins.plugins.sshcredentials.SSHUserPrivateKey;
+import com.cloudbees.plugins.credentials.CredentialsNameProvider;
+import com.cloudbees.plugins.credentials.CredentialsUnavailableException;
+import com.cloudbees.plugins.credentials.NameWith;
+import com.cloudbees.plugins.credentials.common.StandardCertificateCredentials;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+
+import org.jenkinsci.plugins.plaincredentials.StringCredentials;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nonnull;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.util.Secret;
+
+/**
+ * A multi-type credential class backed by AWS Secrets Manager, which detects its type at lookup
+ * time.
+ *
+ * <p>NOTE: The underlying AWS secret must have the necessary format and metadata to be used as a
+ * particular credential type. If these things are not present, the relevant accessor method(s) will
+ * fail at lookup time. (For example, to use the AWS secret as a Jenkins {@link SSHUserPrivateKey},
+ * the secretString must be in private key format, and username metadata must be present in the
+ * secret's tags.)
+ */
+@NameWith(value = AwsCredentials.NameProvider.class)
+abstract class AwsCredentials extends BaseStandardCredentials implements StringCredentials, StandardUsernamePasswordCredentials, SSHUserPrivateKey, StandardCertificateCredentials {
+
+    private static final char[] EMPTY_PASSWORD = {};
+    private static final Secret NONE = Secret.fromString("");
```

Comment:
Super nit, but I'd consider calling this NO_SECRET or NONE_SECRET. That would (arguably) increase code readability IMO

Reply:
Done